### ISSUE TITLE
Overriding an attribute with nil doesn't work during deep merge

### ISSUE THREAD

**Issue by sersut** [chef/chef#1507]
https://tickets.opscode.com/browse/CHEF-4438

Here is a test for the expected behavior:

```
    it "can accept a nil override" do
      @attributes.normal['overriden_by_nil'] = 'cool value!'
      @attributes.override['overriden_by_nil'] = nil
      @attributes['overriden_by_nil'].should == nil
    end
```

https://github.com/michaelglass/chef/blob/override_doesnt_work_for_nil/spec/unit/node/attribute_spec.rb#L355

**Comment by lamont-granquist** [comment]
Yeah, this can't be made to work correctly.  Chef-12 introduced `node.rm('foo')` in order to deal with this.

**Comment by lamont-granquist** [comment]
This will be fixed by #3721
